src/middleware/authMiddleware.js

```diff
@@ -50,11 +53,21 @@ export default async function authMiddleware(req, res, next) {
   }
 
   if (!userId) {
+    // user not found / not authenticated
     res.sendStatus(401);
```

Comment:
we should return here so we don't proceed with the result of the handler - we want to prevent accidentally sending status headers twice in the same response cycle

```suggestion
    return res.sendStatus(401);
```

Reply:
done